crates/dbsp/src/trace/layers/erased/erased_leaf/mod.rs

```diff
@@ -0,0 +1,510 @@
+mod builders;
+mod consumer;
+mod cursor;
+
+use super::{DataVTable, DiffVTable, IntoErasedData, IntoErasedDiff};
+pub use builders::{TypedErasedLeafBuilder, UnorderedTypedLayerBuilder};
+pub use consumer::{TypedLayerConsumer, TypedLayerValues};
+pub use cursor::TypedLayerCursor;
+
+use crate::{
+    algebra::{AddAssignByRef, AddByRef, NegByRef},
+    trace::layers::{advance_erased, Trie},
+    utils::{uninit_vec, DynVec, DynVecVTable},
+    NumEntries,
+};
+use size_of::SizeOf;
+use std::{
+    any::TypeId,
```

Comment:
will be be doing separate compilation for dbsp and the code generated by the compiler?
If so will TypeIds work?
If you need typeids only for values in collections these could be generated by the compiler.

Reply:
there is no guarantee they are consistent between different rust releases. It would be nice if you can generate a more stable id for every type we can store as part of the compiler that would also help with persistence later.